experimental/devices/epd/epd.go

```diff
@@ -0,0 +1,470 @@
+		return err
+	}
+
+	if err := d.sendData([]byte{byte(0xA8)}); err != nil {
+		return err
+	}
+
+	if err := d.sendCommand([]byte{setDummyLinePeriod}); err != nil {
+		return err
+	}
+
+	if err := d.sendData([]byte{byte(0x1A)}); err != nil {
+		return err
+	}
+
+	if err := d.sendCommand([]byte{setGateTime}); err != nil {
+		return err
+	}
+
+	if err := d.sendData([]byte{byte(0x08)}); err != nil {
+		return err
+	}
+
+	if err := d.sendCommand([]byte{dataEntryModeSetting}); err != nil {
+		return err
+	}
+
+	if err := d.sendData([]byte{byte(0x03)}); err != nil {
+		return err
+	}
+
+	if err := d.setLut(Full); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reset can be also used to awaken the device
+func (d *Dev) Reset() {
+	d.rst.Out(gpio.Low)
+	time.Sleep(200 * time.Millisecond)
+	d.rst.Out(gpio.High)
+	time.Sleep(200 * time.Millisecond)
+}
+
+func (d *Dev) setMemoryPointer(x, y int) error {
+	if err := d.sendCommand([]byte{setRAMXAddressCounter}); err != nil {
+		return err
+	}
+
+	// x point must be the multiple of 8 or the last 3 bits will be ignored
+	if err := d.sendData([]byte{byte((x >> 3) & 0xFF)}); err != nil {
+		return err
+	}
+
+	if err := d.sendCommand([]byte{setRAMYAddressCounter}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte(y & 0xFF)}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte((y >> 8) & 0xFF)}); err != nil {
+		return err
+	}
+
+	d.waitUntilIdle()
+
+	return nil
+}
+
+func (d *Dev) waitUntilIdle() {
+	for d.busy.Read() == gpio.High {
+		time.Sleep(100 * time.Millisecond)
+	}
+}
+
+func (d *Dev) setMemoryArea(xStart, yStart, xEnd, yEnd int) error {
+	if err := d.sendCommand([]byte{setRAMXAddressStartEndPosition}); err != nil {
+		return err
+	}
+
+	if err := d.sendData([]byte{byte((xStart >> 3) & 0xFF)}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte((xEnd >> 3) & 0xFF)}); err != nil {
+		return err
+	}
+
+	if err := d.sendCommand([]byte{setRAMYAddressStartEndPosition}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte(yStart & 0xFF)}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte((yStart >> 8) & 0xFF)}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte(yEnd & 0xFF)}); err != nil {
+		return err
+	}
+	if err := d.sendData([]byte{byte((yEnd >> 8) & 0xFF)}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d *Dev) setLut(update PartialUpdate) error {
+	d.update = update
+	lut := d.opts.FullUpdate
+	if d.update == Partial {
+		lut = d.opts.PartialUpdate
+	}
+
+	if err := d.sendCommand([]byte{writeLutRegister}); err != nil {
+		return err
+	}
+
+	for i := range lut {
+		d.sendData([]byte{lut[i]})
+	}
+	return nil
+}
+
+func (d *Dev) sendData(c []byte) error {
+	if err := d.dc.Out(gpio.High); err != nil {
+		return err
+	}
+	return d.c.Tx(c, nil)
+}
+
+func (d *Dev) sendCommand(c []byte) error {
+	if err := d.dc.Out(gpio.Low); err != nil {
+		return err
+	}
+	return d.c.Tx(c, nil)
```

Comment:
I'd recommand to keep dc High after the call, so that consecutive sendData() calls like at line 451 are faster since sendData() doesn't need to touch d.dc at all.

Or keep track of the level in the object.